### PR TITLE
Dev offline check

### PR DESCRIPTION
**Pull request by L-roro**
I started checking for offline functionality but I encountered important bugs on the way that didn't allow me to advance.

When resetting cache and storage data of the app, authentication became impossible. The app was calling Firebase before having logged into it. This pull request adds a workaround to this problem.

I encountered another bug concerning authentication context. Unfortunately, it wasn't a singleton previously (despite the fact that a few methods were trying to modify it). Changed it so that the app shares a unique authentication context.

### CODE REVIEW

File: .idea/compiler.xml
Comment (Lynoxe) [inline .idea/compiler.xml:1]: Avoid pushing this file

File: app/src/main/java/com/github/leuludyha/ibdb/presentation/screen/auth/signup/SignUpScreenViewModel.kt [hunk 1]
```diff
@@ -17,30 +19,55 @@ class SignUpScreenViewModel @Inject constructor(
 
     private var walkThroughCompleted by mutableStateOf(false)
 
+    //TODO: All methods here are implemented using SharedPreferences. I believe this will be changed
+    // further in development, as Dylan intended to implement userPreferences in a different way.
+    // This is just a temporary way of doing it.
+
```
Comment (Lynoxe) [inline app/src/main/java/com/github/leuludyha/ibdb/presentation/screen/auth/signup/SignUpScreenViewModel.kt:25]: Actually, I meant to use `DataStore` which is the newer equivalent of shared preferences. I don't think it really matters which one we use as long as it works.

**Review by Lynoxe — APPROVED** [review]

File: .idea/compiler.xml
Comment (L-roro) [reply .idea/compiler.xml:1]: Yep, sorry

**Comment by L-roro** [comment]
I made this pull request as a messy hotfix for those that had problems with the authentication calling firebase before the user had logged in. Apparently no one truly needed it, so I will just cancel it, as it is a disaster. I will reopen this in the branch [dev-offline-check-fixed](https://github.com/LEuLuDyHa/app/tree/dev-offline-check-fixed) with something a bit more complete.

**Pull request closed without merging**